torch_xla/csrc/view.cpp

```diff
@@ -0,0 +1,103 @@
+#include "torch_xla/csrc/view.h"
+
+#include <algorithm>
+#include <functional>
+#include <numeric>
+
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/ops/generic_slice.h"
+#include "torch_xla/csrc/ops/update_slice.h"
+#include "torch_xla/csrc/ops/view.h"
+
+namespace torch_xla {
+namespace {
+
+bool IsNarrow(const ViewInfo& view_info) {
```

Comment:
I'm not sure I like the fact that the only way to distinguish between what's two distinct operators (view and narrow) is so implicit. Could we store the kind symbol in the `ViewInfo` as well?

Reply:
So far we only support narrow and view/reshape, and this is what drives the two different update policies (require dyn-update-slice vs reshape).